addons/pkg-maven/common/src/main/java/org/commonjava/indy/pkg/maven/content/MavenMetadataGenerator.java

```diff
@@ -612,11 +612,10 @@ private void putToMetadataCache( StoreKey key, String toMergePath, MetadataInfo
             if ( cacheMap == null )
             {
                 cacheMap = new HashMap<>();
+                versionMetadataCache.put( key, cacheMap );
             }
 
             cacheMap.put( toMergePath, meta );
-            versionMetadataCache.put( key, cacheMap );
```

Comment:
The put() should be in the curly bracket. This causes many unnecessary cache operations.

Reply:
If the cacheMap already exists in the verisonMetadataCache and we only add a value to it, will Infinispan see this as a cache update? If we start clustering this cache, will the map with the new value in it be visible to other nodes in the cluster?

Reply:
Not sure. I'll test it.